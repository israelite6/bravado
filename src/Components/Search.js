import React from 'react';
import PropTypes from 'prop-types';
import {View, StyleSheet, TextInput} from 'react-native';
import Icon from 'react-native-vector-icons/AntDesign';
import {useDebouncedCallback} from 'use-debounce';

/**
 *
 * @param {string} defaultValue
 * @param {*} onChange - callback function for search
 * @returns
 */
export default function Search({defaultValue, onChange, ...props}) {
  const debounced = useDebouncedCallback(
    // function
    value => {
      onChange(value);
    },
    // delay in ms
    300,
  );
  return (
    <View style={styles.root}>
      <Icon name="search1" size={12} style={styles.icon} />
      <TextInput
        style={styles.textInput}
        defaultValue={defaultValue}
        onChangeText={debounced}
      />
    </View>
  );
}

Search.propTypes = {
  defaultValue: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#FFFFFF',
    paddingVertical: 10,
    paddingHorizontal: 12,
    borderRadius: 4,
  },
  icon: {
    color: '#B3B3B3',
    paddingRight: 9,
  },
  textInput: {
    flexGrow: 1,
    fontSize: 14,
    paddingVertical: 0,
  },
});
