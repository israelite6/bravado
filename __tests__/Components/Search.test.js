/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../../src/Components/Search';

it('Avatar component renders correctly', () => {
  const tree = renderer.create(<Search onChange={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
