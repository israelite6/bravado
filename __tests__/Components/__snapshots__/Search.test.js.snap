// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 4,
      "flexDirection": "row",
      "paddingHorizontal": 12,
      "paddingVertical": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 10,
        },
        Object {
          "color": "#B3B3B3",
          "paddingRight": 9,
        },
        Object {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î™°
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "flexGrow": 1,
        "fontSize": 14,
        "paddingVertical": 0,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
